### PR TITLE
nuttx: fix  coverity check error 

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
## Summary

Patch 1:  fix deadcode in pthread_mutexinit
Patch 2: timer:settime: check return value of clock_abstime2ticks

## Impact

No. bug fix

## Testing

Testing in armv8-m board

### CODE REVIEW

File: sched/pthread/pthread_mutexinit.c [hunk 1]
```diff
@@ -124,11 +117,20 @@ int pthread_mutex_init(FAR pthread_mutex_t *mutex,
...
 
 #ifndef CONFIG_PTHREAD_MUTEX_UNSAFE
-      /* Initial internal fields of the mutex */
 
       mutex->flink  = NULL;
+
+#ifdef CONFIG_PTHREAD_MUTEX_BOTH
+      /* Initial internal fields of the mutex */
+
       mutex->flags  = (robust == PTHREAD_MUTEX_ROBUST ?
```
Comment (pkarashchenko) [inline sched/pthread/pthread_mutexinit.c:126]: If you removed `uint8_t robust` at lines 72 and 74, then what is checked here?

Comment (zhuyanlinzyl) [reply sched/pthread/pthread_mutexinit.c:131]: Yes, I add uint8_t robust when CONFIG_PTHREAD_MUTEX_BOTH is enable.

Comment (zhuyanlinzyl) [reply sched/pthread/pthread_mutexinit.c:131]: @pkarashchenko  please review again

Comment (pkarashchenko) [reply sched/pthread/pthread_mutexinit.c:131]: what is the exact error reported here by coverity?

File: sched/pthread/pthread_mutexinit.c [hunk 2]
```diff
@@ -68,11 +68,9 @@ int pthread_mutex_init(FAR pthread_mutex_t *mutex,
 #  endif
 #endif
 #ifndef CONFIG_PTHREAD_MUTEX_UNSAFE
-#ifdef CONFIG_PTHREAD_MUTEX_DEFAULT_UNSAFE
-  uint8_t robust = PTHREAD_MUTEX_STALLED;
-#else
-  uint8_t robust = PTHREAD_MUTEX_ROBUST;
-#endif
+#  ifdef CONFIG_PTHREAD_MUTEX_BOTH
+  uint8_t robust;
```
Comment (xiaoxiang781216) [inline sched/pthread/pthread_mutexinit.c:72]: may not initialize if attr is NULL.

File: sched/pthread/pthread_mutexinit.c [hunk 3]
```diff
@@ -97,7 +97,7 @@ int pthread_mutex_init(FAR pthread_mutex_t *mutex,
           type    = attr->type;
 #endif
 #ifdef CONFIG_PTHREAD_MUTEX_BOTH
-          robust  = attr->robust;
+          flags  = attr->robust == PTHREAD_MUTEX_ROBUST ? _PTHREAD_MFLAGS_ROBUST : 0;
```
Comment (pkarashchenko) [inline sched/pthread/pthread_mutexinit.c:100]:
```suggestion
          flags   = attr->robust == PTHREAD_MUTEX_ROBUST ? _PTHREAD_MFLAGS_ROBUST : 0;
```

File: sched/pthread/pthread_mutexinit.c [hunk 4]
```diff
@@ -69,9 +69,9 @@ int pthread_mutex_init(FAR pthread_mutex_t *mutex,
 #endif
 #ifndef CONFIG_PTHREAD_MUTEX_UNSAFE
 #ifdef CONFIG_PTHREAD_MUTEX_DEFAULT_UNSAFE
-  uint8_t robust = PTHREAD_MUTEX_STALLED;
+  uint8_t flags = 0;
 #else
-  uint8_t robust = PTHREAD_MUTEX_ROBUST;
+  uint8_t flags = _PTHREAD_MFLAGS_ROBUST;
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply sched/pthread/pthread_mutexinit.c:74]: Where it is inconsistent? you mean the error check or variable naming.

[the remark this answers is not in the document]
Comment (pkarashchenko) [reply sched/pthread/pthread_mutexinit.c:74]: Here: `robust == PTHREAD_MUTEX_STALLED || robust == _PTHREAD_MFLAGS_ROBUST`
It checks `robust` against `_PTHREAD_MFLAGS_ROBUST` and not `PTHREAD_MUTEX_ROBUST`

[the remark this answers is not in the document]
Comment (pkarashchenko) [reply sched/pthread/pthread_mutexinit.c:74]: Those two in fact are both 1s, so code works

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply sched/pthread/pthread_mutexinit.c:74]: Ok, pthread_mutexattr_setrobust check the wrong macro, @zhuyanlinzyl please change "robust == _PTHREAD_MFLAGS_ROBUST" to "robust == PTHREAD_MUTEX_ROBUST"

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**